CUDADataFormats/SiStripCluster/interface/SiStripClustersCUDA.h

```diff
@@ -44,11 +44,54 @@ class SiStripClustersCUDA {
     stripgpu::stripId_t *firstStrip_;
     bool *trueCluster_;
     float *barycenter_;
-    int nClusters_;
+
   };
 
   DeviceView *view() const { return view_d.get(); }
 
+  class GlobalDeviceView {
+  public:
+
+//    __device__ __forceinline__ float local_xx(int i) const  { return __ldg(local_xx_ + i); }
+//    __device__ __forceinline__ float local_xy(int i) const  { return __ldg(local_xy_ + i); }
+//    __device__ __forceinline__ float local_yy(int i) const  { return __ldg(local_yy_ + i); }
+    __device__ __forceinline__ float global_x(int i) const  { return __ldg(global_x_ + i); }
+    __device__ __forceinline__ float global_y(int i) const  { return __ldg(global_y_ + i); }
+    __device__ __forceinline__ float global_z(int i) const  { return __ldg(global_z_ + i); }
+
+    __device__ __forceinline__ float global_xx(int i) const  { return __ldg(global_xx_ + i); }
+    __device__ __forceinline__ float global_xy(int i) const  { return __ldg(global_xy_ + i); }
+    __device__ __forceinline__ float global_xz(int i) const  { return __ldg(global_xz_ + i); }
+    __device__ __forceinline__ float global_yy(int i) const  { return __ldg(global_yy_ + i); }
+    __device__ __forceinline__ float global_yz(int i) const  { return __ldg(global_yz_ + i); }
+    __device__ __forceinline__ float global_zz(int i) const  { return __ldg(global_zz_ + i); }
+
+    __device__ __forceinline__ short layer(int i) const  { return __ldg(layer_ + i); }
+
+    friend SiStripClustersCUDA;
+
+    //   private:
+    int nClusters_;
+
+//    float *local_xx_;
+//    float *local_xy_;
+//    float *local_yy_;
+    float *global_x_;
+    float *global_y_;
+    float *global_z_;
+
+    float *global_xx_;
+    float *global_xy_;
+    float *global_xz_;
+    float *global_yy_;
+    float *global_yz_;
+    float *global_zz_;
+
+    short *layer_;
+  };
+
+  GlobalDeviceView *gview() const { return gview_d.get(); }
+
```

Comment:
This should be a separate data format.  A data format should do one thing, not mix different products from different producers.